### PR TITLE
proposal: Avoid negative variable names

### PR DESCRIPTION
**Pull request by strashkov**
https://medium.com/dfind-consulting/about-writing-positive-and-negative-variable-names-b42328d51e9c

### CODE REVIEW

File: src/general.md [hunk 1]
```diff
@@ -117,6 +117,22 @@ In the UI, the language should be local. For the UK, Singapore, and HK, it's Bri
    const failedAuthorizationPayload = 'authorization_failed';
    ```
 
+1. Avoid negative variable names.
+
+   ```typescript
+   // bad
+
+   if (!cantDoIt) {
```
Comment (dmlyapin) [inline src/general.md:125]: can you suggest a real sample (the reason why you are adding this rule)

Comment (strashkov) [reply src/general.md:125]: the reason is negative names can lead to double negation and this increases the cognitive load when reading the code
this is a fairly common principle when writing code
[here](https://medium.com/dfind-consulting/about-writing-positive-and-negative-variable-names-b42328d51e9c) you can find examples and a good summary

Comment (strashkov) [reply src/general.md:125]: sometimes I face negative names in our repos and I want to have arguments to ask developers not to do that

Comment (dmlyapin) [reply src/general.md:125]: I understand the problem, I just mean let's use a real sample here, not "cantDoIt"

> sometimes I face negative names in our repos

let's take one of them

Comment (strashkov) [reply src/general.md:125]: recent one
https://github.com/OsomePteLtd/sdk/pull/2721#discussion_r1121514578

Comment (dmlyapin) [reply src/general.md:125]: Let's update the bad / good sample?

```
// bad

noDiscountAdjustment

// good

isDiscountAdjustmentDisabled
```

(btw, is not i`sDiscountAdjustmentDisabled` also **negative**?)

Comment (dmlyapin) [reply src/general.md:125]: I suggest to avoid synthetic sample in principles like `cantDoIt` and use only real samples from PRs.

**Review by dmlyapin — COMMENTED** [review]
https://github.com/OsomePteLtd/principles/pull/236/files#r1124320658

File: src/general.md (hunk 1 above)
Comment (strashkov) [reply src/general.md]: > (btw, is not isDiscountAdjustmentDisabled also negative?)

disabled, inactive, etc are not negative according to the English and do not lead to double negation (`it is not disabled` looks good grammatically)

**Review by dmlyapin — APPROVED** [review]

**Pull request merged**